src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/DependencyAnalysis/ReadyToRun/ManifestMetadataTableNode.cs

```diff
@@ -204,7 +226,7 @@ public override ObjectData GetData(NodeFactory factory, bool relocsOnly = false)
                     name: metadataBuilder.GetOrAddString(assemblyName.Name),
                     version: assemblyName.Version,
                     culture: metadataBuilder.GetOrAddString(assemblyName.CultureName),
-                    publicKeyOrToken: metadataBuilder.GetOrAddBlob(publicKeyOrToken),
+                    publicKeyOrToken: publicKeyOrToken == null ? default : metadataBuilder.GetOrAddBlob(publicKeyOrToken),
                     flags: assemblyFlags,
                     hashValue: default(BlobHandle) /* TODO */);
```

Comment:
Why don't we just use the `hashValue` here to store the MVID instead of a new node type?

Reply:
Unless of course we have plans to use hashValue for something else, but the list of assemblies written in this MD blob seem to only be used for the version bubble stuff.

Reply:
This table only has references to assemblies that don't exist in the original AssemblyRef table. There's a [complex scheme](https://github.com/dotnet/runtime/blob/1bbd51d56a8e4e2895a9b707cd0551ee0b28e253/docs/design/coreclr/botr/readytorun-format.md#readytorun_section_manifest_metadata) of how to make sense of it.

We can't modify the original AssemblyRef table because it's visible to user code. We would need an alternative storage mechanism for those entries anyway - people do weird stuff with user visible parts of these things - e.g. I've seen an app that used the assembly's public key to encrypt network traffic 😱).

Reply:
Ok Thanks Michal for the clarification